src/app/server/Dnssd.cpp

```diff
@@ -472,44 +475,35 @@ void DnssdServer::StartServer(Dnssd::CommissioningMode mode)
         ChipLogError(Discovery, "Failed to advertise operational node: %s", chip::ErrorStr(err));
     }
 
-    if (HaveOperationalCredentials())
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY
```

Comment:
That config value controls whether the device should advertise via dns-sd _when it's not yet commissioned at all_.  That's why it was only checked when `HaveOperationalCredentials()` was false.  

If the device commissioned and a commissioning window was via the relevant cluster commands, the device _has_ to advertise via dns-sd per the spec, but after this change it won't do that if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY is false.

Was opening a commissioning window and commissioning a device into a second fabric tested with this PR on a Thread device?

Reply:
Ok, so maybe we should drop `CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY` at all? Since any case where the commissioning mode is 0 is extended discovery for which we have another switch. It was tested on Thread platforms, because all Thread platforms I'm aware of set this to true. Not sure it's useful anyhow.

Reply:
I think the intent is that some devices do want to advertise on the IP network (if they have one) when they are uncommissioned and in commissioning mode and others do not want to so advertise.  That choice is what CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY is meant to handle.

I personally have no problem getting rid of CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY and just saying that if you're in commissioning mode and not yet commissioned you should advertise across all the ways you can be discovered including dns-sd....

@chrisdecenzo Any idea why we have this config option?

Reply:
Yeah, let's hear from @chrisdecenzo. It just thought it could be simplified since I haven't found anything in the spec suggesting that a different timeout should be applied to the case where the device is commissioned and where it's uncommissioned yet. The spec only mentions CM=0. But I can revert that part if this can be useful.

Reply:
it looks like CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY was introduced by @mkardous-silabs and its only 0 when CHIP_DEVICE_CONFIG_ENABLE_THREAD. 

I don't understand why Thread would need DNS-SD commissionable discovery to be disabled when the device has not yet been commissioned, but maybe it causes errors because the device is not yet on a network.

I'm fine with removing CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY and go with the logic where it was always 1, but please verify this works on Thread devices.

Reply:
Just to clarify, the PR you are referring to did not add any logic. I just renamed a define from  `CHIP_DEVICE_CONFIG_ENABLE_UNPROVISIONED_MDNS` to `CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY`.
https://github.com/project-chip/connectedhomeip/pull/9111#discussion_r692413237

Reply:
Ok, I guess I set this to 0 for Thread platforms back in time, because some versions of OpenThread had different issues with SRP and I wanted all platform maintainers to verify it works correctly before it gets enabled. But now it looks like all platforms have extended discovery enabled if and only if they have `CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY` set, too, so I think we can get rid of the latter setting.